internal/buffer/line_array.go

```diff
@@ -46,9 +46,8 @@ type searchState struct {
 type Line struct {
 	data []byte
 
-	state       highlight.State
-	match       highlight.LineMatch
-	rehighlight bool
```

Comment:
Just a note: I can actually imagine a use for it: if we move all the highlighting to the async goroutine (to fix the other race we were talking about: the race between initial highlighting in the async goroutine and rehighlighting in the main goroutine; i.e. to ensure that initial highlighting and rehighlighting for the given line are done in the correct order), then the main goroutine may call `SetRehighlight()` to let the async goroutine know that the given line requires rehighlighting.

But it's probably better to do such notifications using a channel, not a mutex-locked shared variable. So I'm ok with removing this.